### PR TITLE
Remove redundant Futures tests (#24095)

### LINKED ISSUE #24095: Remove Future tests, since Future is now Scala's "own" Future
Observed by @2m 

> Ooh. This touched some old tests. :) Most of the tests in FutureSpec.scala are now a part of scala tests that can be found in https://github.com/scala/scala/blob/2.13.x/test/files/jvm/future-spec/FutureTests.scala and https://github.com/scala/scala/blob/2.13.x/test/files/jvm/future-spec/PromiseTests.scala
> 
> It would be a good idea to remove these duplicated tests from Akka all together.

### PR DESCRIPTION
**Pull request by catalin-ursachi**
Fixes #24095.

I've removed every test that was obviously irrelevant to Akka. The remaining ones look like they're covered by the tests in `AskSpec`, though; should I just delete FutureSpec altogether?

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by catalin-ursachi** [comment]
The error looks spurious & unrelated to the changes, as far as I can tell?

File: akka-actor-tests/src/test/scala/akka/pattern/AskSpec.scala [hunk 1]
```diff
@@ -42,7 +42,7 @@ class AskSpec extends AkkaSpec {
       f.isCompleted should ===(true)
       f.value.get match {
         case Failure(_: AskTimeoutException) ⇒
-        case v                               ⇒ fail(v + " was not Left(AskTimeoutException)")
+        case v                               ⇒ fail(v + " was not Failure(AskTimeoutException)")
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/pattern/AskSpec.scala]: Good

File: akka-actor-tests/src/test/scala/akka/dispatch/FutureSpec.scala [hunk 2]
```diff
@@ -1,215 +1,50 @@
...
-    try Await.ready(awaitable, atMost) catch {
-      case t: TimeoutException ⇒ throw t
-      case e if NonFatal(e)    ⇒ awaitable //swallow
-    }
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.language.postfixOps
+import scala.reflect.ClassTag
+import scala.util.{ Failure, Success }
 
+object FutureSpec {
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/dispatch/FutureSpec.scala]: I think you can remove the entire test file as ask has tests in `AskSpec`

**Review by johanandren — CHANGES_REQUESTED** [review]

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
OK TO TEST

**Comment by patriknw** [comment]
PLS BUILD

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**